include/capstone/riscv.h

```diff
@@ -0,0 +1,428 @@
+} riscv_op_type;
+
+// Instruction's operand referring to memory
+// This is associated with RISCV_OP_MEM operand type above
+typedef struct riscv_op_mem {
+	unsigned int base;	// base register
+	int64_t disp;	// displacement/offset value
+} riscv_op_mem;
+
+// Instruction operand
+typedef struct cs_riscv_op {
+	riscv_op_type type;	// operand type
+	union {
+		unsigned int reg;	// register value for REG operand
+		int64_t imm;		// immediate value for IMM operand
+		riscv_op_mem mem;	// base/disp value for MEM operand
+	};
+} cs_riscv_op;
+
+// Instruction structure
+typedef struct cs_riscv {
+	// Number of operands of this instruction, 
+	// or 0 when instruction has no operand.
+	uint8_t op_count;
+	cs_riscv_op operands[8]; // operands for this instruction.
+} cs_riscv;
+
+//> RISCV registers
+typedef enum riscv_reg {
+	RISCV_REG_INVALID = 0,
+	//> General purpose registers
+	RISCV_REG_X0, 
+	RISCV_REG_X1, 
+	RISCV_REG_X2, 
+	RISCV_REG_X3, 
+	RISCV_REG_X4, 
+	RISCV_REG_X5, 
+	RISCV_REG_X6, 
+	RISCV_REG_X7, 
+	RISCV_REG_X8, 
+	RISCV_REG_X9, 
+	RISCV_REG_X10,
+	RISCV_REG_X11,
+	RISCV_REG_X12,
+	RISCV_REG_X13,
+	RISCV_REG_X14,
+	RISCV_REG_X15,
+	RISCV_REG_X16,
+	RISCV_REG_X17,
+	RISCV_REG_X18,
+	RISCV_REG_X19,
+	RISCV_REG_X20,
+	RISCV_REG_X21,
+	RISCV_REG_X22,
+	RISCV_REG_X23,
+	RISCV_REG_X24,
+	RISCV_REG_X25,
+	RISCV_REG_X26,
+	RISCV_REG_X27,
+	RISCV_REG_X28,
+	RISCV_REG_X29,
+	RISCV_REG_X30,
+	RISCV_REG_X31,
+
+	//> Single-precision floating-point registers
+	RISCV_REG_F0_32, 
+	RISCV_REG_F1_32, 
+	RISCV_REG_F2_32, 
+	RISCV_REG_F3_32, 
+	RISCV_REG_F4_32, 
+	RISCV_REG_F5_32, 
+	RISCV_REG_F6_32, 
+	RISCV_REG_F7_32, 
+	RISCV_REG_F8_32, 
+	RISCV_REG_F9_32,
+	RISCV_REG_F10_32, 
+	RISCV_REG_F11_32, 
+	RISCV_REG_F12_32, 
+	RISCV_REG_F13_32, 
+	RISCV_REG_F14_32, 
+	RISCV_REG_F15_32, 
+	RISCV_REG_F16_32, 
+	RISCV_REG_F17_32, 
+	RISCV_REG_F18_32, 
+	RISCV_REG_F19_32,  
+	RISCV_REG_F20_32,
+	RISCV_REG_F21_32,
+	RISCV_REG_F22_32,
+	RISCV_REG_F23_32,
+	RISCV_REG_F24_32,
+	RISCV_REG_F25_32,
+	RISCV_REG_F26_32,
+	RISCV_REG_F27_32,
+	RISCV_REG_F28_32,
+	RISCV_REG_F29_32,
+	RISCV_REG_F30_32,
+	RISCV_REG_F31_32,
+
+	//> Double-precision floating-point registers
+	RISCV_REG_F0_64, 
+	RISCV_REG_F1_64, 
+	RISCV_REG_F2_64, 
+	RISCV_REG_F3_64, 
+	RISCV_REG_F4_64, 
+	RISCV_REG_F5_64, 
+	RISCV_REG_F6_64, 
+	RISCV_REG_F7_64, 
+	RISCV_REG_F8_64, 
+	RISCV_REG_F9_64,
+	RISCV_REG_F10_64, 
+	RISCV_REG_F11_64, 
+	RISCV_REG_F12_64, 
+	RISCV_REG_F13_64, 
+	RISCV_REG_F14_64, 
+	RISCV_REG_F15_64, 
+	RISCV_REG_F16_64, 
+	RISCV_REG_F17_64, 
+	RISCV_REG_F18_64, 
+	RISCV_REG_F19_64,  
+	RISCV_REG_F20_64,
+	RISCV_REG_F21_64,
+	RISCV_REG_F22_64,
+	RISCV_REG_F23_64,
+	RISCV_REG_F24_64,
+	RISCV_REG_F25_64,
+	RISCV_REG_F26_64,
+	RISCV_REG_F27_64,
+	RISCV_REG_F28_64,
+	RISCV_REG_F29_64,
+	RISCV_REG_F30_64,
+	RISCV_REG_F31_64,
+
+	RISCV_REG_ENDING,	// <-- mark the end of the list or registers
+} riscv_reg;
+
+//> RISCV instruction
+typedef enum riscv_insn {
+	RISCV_INS_INVALID = 0,
```

Comment:
this file is not indented with tabs yet?

Reply:
Fixed